src/zepben/evolve/model/busbranch/bus_branch.py

```diff
@@ -926,17 +923,14 @@ async def add_to_group(t: Terminal, _):
 
 
 def _queue_terminals_across_negligible_impedance(
-    tg: TerminalGrouping[ConductingEquipment],
     has_negligible_impedance: Callable[[ConductingEquipment], bool]
 ):
     def queue_next(terminal: Terminal, traversal: Traversal[Terminal]):
         if terminal.connectivity_node is not None:
-            traversal.process_queue.extend({ot for ot in terminal.connectivity_node.terminals
-                                            if ot != terminal and ot not in tg.terminals()})
+            traversal.process_queue.extend(ot for ot in terminal.connectivity_node.terminals if ot != terminal)
 
         if has_negligible_impedance(terminal.conducting_equipment):
-            traversal.process_queue.extend({ot for ot in terminal.conducting_equipment.terminals
-                                            if ot != terminal and ot not in tg.terminals()})
+            traversal.process_queue.extend(ot for ot in terminal.conducting_equipment.terminals if ot != terminal)
```

Comment:
This change alone speeds up bus branch creation by ~100%. The traversal will handle duplicate entries in the process queue.